From 3928b1320ddc36823f55f5e343a2737b7b1e6959 Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Fri, 5 Jun 2015 13:56:33 +0800
Subject: [PATCH 1/5] Public Headers: Allow Build On Visual Studio

Ensure that we also have import/export decorations for the prototypes, as
Visual Studio does not like such decorations on the function definitions
without the prototypes decorated as well, when the function prototypes
exist.
---
 include/epoxy/gl.h  | 6 +++---
 include/epoxy/wgl.h | 4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/epoxy/gl.h b/include/epoxy/gl.h
index bbc06c0..0be3a66 100644
--- a/include/epoxy/gl.h
+++ b/include/epoxy/gl.h
@@ -93,9 +93,9 @@ extern "C" {
 
 #include "epoxy/gl_generated.h"
 
-bool epoxy_has_gl_extension(const char *extension);
-bool epoxy_is_desktop_gl(void);
-int epoxy_gl_version(void);
+EPOXY_IMPORTEXPORT bool epoxy_has_gl_extension(const char *extension);
+EPOXY_IMPORTEXPORT bool epoxy_is_desktop_gl(void);
+EPOXY_IMPORTEXPORT int epoxy_gl_version(void);
 
 #ifdef __cplusplus
 } /* extern "C" */
diff --git a/include/epoxy/wgl.h b/include/epoxy/wgl.h
index 2c4682f..fdd7b4f 100644
--- a/include/epoxy/wgl.h
+++ b/include/epoxy/wgl.h
@@ -54,8 +54,8 @@ extern "C" {
 
 #include "epoxy/wgl_generated.h"
 
-bool epoxy_has_wgl_extension(HDC hdc, const char *extension);
-void epoxy_handle_external_wglMakeCurrent(void);
+EPOXY_IMPORTEXPORT bool epoxy_has_wgl_extension(HDC hdc, const char *extension);
+EPOXY_IMPORTEXPORT void epoxy_handle_external_wglMakeCurrent(void);
 
 #ifdef __cplusplus
 } /* extern "C" */
-- 
2.3.5.windows.8

